File: matrix.js
// take an n by m matrix and tranpose it to be m by n
console.log("Enter your matrix row by row seperated by spaces and each row by a new line: ");
const prompt=require("prompt-sync")({sigint:true}); 

let UserInput = prompt();
const Matrix = []
while(UserInput !== 'done'){
   // console.log(UserInput);
    //add the value into an array?
    Matrix.push(UserInput);
    UserInput = prompt();
    //console.log(Matrix[0][0])
}


//console.log(UserInput);

// Iterate through UserInput

//console.log(Matrix);
const FinalMatrix = [];
for(let i = 0; i < Matrix[0].length; i++){
    let row = [];
    for (let j = 0; j < Matrix.length;j++){
        //console.log(UserInput[i]);
         if (Matrix[j][i] !== " "){
            row.push(Matrix[j][i]);
            //console.log(Matrix[j][i]);
        }

    }
    FinalMatrix.push(row);
    
}

FinalMatrix.forEach(function(row){
    row.forEach(function(element){
        console.log(element);
    });

});
console.log(`Your final Matrix is: ${FinalMatrix}`);

File: matrix.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

int main() {
    vector<vector<string>> matrix;
    cout << "Give your Matrix with each item in the row separated by spaces, and each row separated by a newline (TYPE q WHEN DONE)" << endl;
    string line;

    // Parse the input and build the matrix
    while (getline(cin, line)) {
        if (line == "q" || line == "quit"|| line == "Quit") break;
        vector<string> row;
        stringstream ss(line);
        string item;
        while (ss >> item) {
            row.push_back(item);
        }
        matrix.push_back(row);
    }

    // Transpose the matrix
    vector<vector<string>> transposed(matrix[0].size(), vector<string>(matrix.size()));
    for (size_t i = 0; i < matrix.size(); ++i) {
        for (size_t j = 0; j < matrix[i].size(); ++j) {
            transposed[j][i] = matrix[i][j];
        }
    }

    // Print the transposed matrix
    cout << "Transposed Matrix:" << endl;
    for (size_t i = 0; i < transposed.size(); ++i) {
        for (size_t j = 0; j < transposed[i].size(); ++j) {
            cout << transposed[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}
